### PR TITLE
Setup Cuda testing on Jenkins

### PR DESCRIPTION
**Pull request by dalg24**
Note that all existing PRs will need to be rebased on these changes for Jenkins build to work.

### CODE REVIEW

**Comment by dalg24** [comment]
@sslattery The AoSoA did not pass.  I suggest we fix this in another PR.
As you see if you follow [this link](https://cloud.cees.ornl.gov/jenkins-ci/job/Cabana/2/) the build hooks are in place and seem to work.  Some adjustment might be needed in the Jenkins configuration (possibly the white list for the trigger) but I granted you and Christoph permissions to edit it.

**Comment by dalg24** [comment]
> Yet another CI system? Didn’t we wanted to install a gitlab runner on that machine?

GitLab does not mirror the PRs.

**Comment by junghans** [comment]
Their idea of verification is that a developer review external PRs and push the sha on a local branch, which will trigger gitlab CI!

**Comment by junghans** [comment]
> Their idea of verification is that a developer review external PRs and push the sha on a local branch, which will trigger gitlab CI!

It seems this broke like a month ago.

**Comment by junghans** [comment]
We are back in business!

**Comment by junghans** [comment]
And it seems gitlab.com triggers pull requests automatically now!

**Comment by dalg24** [comment]
Closing per https://github.com/ECP-copa/Cabana/issues/19#issuecomment-470551415 and disabling the job on the Jenkins server

**Comment by junghans** [comment]
https://gitlab.com/gitlab-org/gitlab-ee/issues/5667 confuses me. I am not sure why your fork got triggered

**Comment by dalg24** [comment]
Reopening since #79 is not able to address #19 at this time

**Comment by dalg24** [comment]
Merged #80 that fixed the test failure into that branch.  The build did not trigger on Jenkins though.  Will look into it.

**Comment by dalg24** [comment]
Went to the GitHub webhooks settings and redelivered the payload.  Build passed.  Let me do a rebase or something similar to re-trigger and be 100% sure it works now.

**Comment by dalg24** [comment]
Build triggered.  This is ready to merged.

**Comment by dalg24** [comment]
@junghans @sslattery NOTE the CI is only setup for PRs at this time.

File: scripts/jenkins/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM nvidia/cuda:10.0-devel-ubuntu18.04
+RUN apt-get update && apt-get install -y \
+        bc \
+        wget \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install CMake
+RUN export CMAKE_VERSION=3.13.4 && \
```
Comment (dalg24) [inline scripts/jenkins/Dockerfile]: Cross referencing https://github.com/ECP-copa/Cabana/pull/79/files#r264784631

@junghans are you satisfied with my reply?

Comment (junghans) [reply scripts/jenkins/Dockerfile]: Fine with me!

**Pull request merged**